src/lib/OS/SyscallHandler.cc

```diff
@@ -391,50 +388,31 @@ int64_t SyscallHandler::openat(int64_t dfd, const std::string& filename,
   int64_t dirfd = SyscallHandler::getDirFd(dfd, filename);
   if (dirfd == -1) return EBADF;
 
-  // Pass call through to host
-  int64_t hfd = ::openat(dirfd, new_pathname.c_str(), newFlags, mode);
-  if (hfd < 0) {
-    return hfd;
-  }
-
-  std::shared_ptr<Process> proc = processes_.find(0)->second;
-
-  // Allocate virtual file descriptor and map to host file descriptor
-  int64_t vfd;
-  if (!proc->freeFileDescriptors_.empty()) {
-    // Take virtual descriptor from free pool
-    auto first = proc->freeFileDescriptors_.begin();
-    vfd = proc->freeFileDescriptors_.extract(first).value();
-    proc->fileDescriptorTable_[vfd] = hfd;
-  } else {
-    // Extend file descriptor table for a new virtual descriptor
-    vfd = proc->fileDescriptorTable_.size();
-    proc->fileDescriptorTable_.push_back(hfd);
-  }
-
-  return vfd;
+  std::shared_ptr<Process> proc = os_->getProcess(0);
```

Comment:
`getProcess` returns type of `const shared_ptr<Process>&`

Reply:
done